### PR TITLE
Implementing a custom zoom in and zoom out imageview in android with kotlin

### LINKED ISSUE #4263: [Android] Implementing a Custom Zoom in and Zoom out ImageView in Android with Kotlin
#### [Android] Implementing a Custom Zoom in and Zoom out ImageView in Android with Kotlin

In android zoom in and zoom out are used to change the size and appearance of the image over particular interval of time.These feature provide a better look and feel for the designed application.

In this tutorial we'll learn how to create a custom zoom in and zoom out which is not covered in most available article and learn on the various zoom control attributes on android. Also we'll look at  how they are implemented under several circumstances and  learn on the various listeners used to handle zooming effects in android.

#### Key takeaways 
- What are the importance of including Zoom capabilities to your Android Application
- How does zoom in and zoom out works.
- Various methods used in zoom control.
- Different type of listeners used in the implementation of zooming.
- Attributes of zoom control.
- How to create a custom zoom in and zoom out ImageView.

#### Article quality
The article  differ from the existing article in that it explain a way of designing a custom zoom in and zoom out image view which in most available article  this features are  explained with the help of a built-in libraries.
The article will be short and precise making it more user friendly and readable to both beginners and experienced Android Developers. Readers tend to get bored with longer articles which adds alot of boilerplate codes. 

#### References
N/A

### PR DESCRIPTION
**Pull request by brandy-kay**
This fixes  #4263

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check done

File: content/articles/implementing-a-custom-zoom-in-and-zoom-out-imageview-in-android-with-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+- `View.OnTouchListener`
+- `GestureDetector.OnGestureListener`
+- `GestureDetector.OnDoubleTapListener`
+
+Within the created class ensure you have declared the following variables that would aid in the creation of these zooming controls.
+
+```kotlin
+    var myMatrix: Matrix? = null
+    private var matrixValue: FloatArray? = null
+    var mode = NONE
```
Comment (Ericgacoki) [inline content/articles/implementing-a-custom-zoom-in-and-zoom-out-imageview-in-android-with-kotlin/index.md:28]: Indent your code properly.

File: content/articles/implementing-a-custom-zoom-in-and-zoom-out-imageview-in-android-with-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+private inner class ScalingListener : ScaleGestureDetector.SimpleOnScaleGestureListener(){
+    ...
+}
+```
+
+This class overrides two methods that are `onScaleBegin` and `onScale` which contains the general implementation of the `ScaleGestureDetector` interface.
+
+`onScaleBegin` shows the mode level of the zooming which are ought to be done.
+```kotlin
+        override fun onScaleBegin(detector: ScaleGestureDetector?): Boolean {
```
Comment (Ericgacoki) [inline content/articles/implementing-a-custom-zoom-in-and-zoom-out-imageview-in-android-with-kotlin/index.md:70]: The same to this and the rest where applicable.

Comment (brandy-kay) [reply content/articles/implementing-a-custom-zoom-in-and-zoom-out-imageview-in-android-with-kotlin/index.md:86]: Hello @Ericgacoki  have made made the changes

**Comment by Ericgacoki** [comment]
Hey @brandy-kay, I've seen a great improvement from your previous article(s). Keep the fire burning :fire:

**Comment by brandy-kay** [comment]
thanks Eric

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
Overall very nice work 🚀 

Made some minor edits - please review

**Pull request merged**